### PR TITLE
Create Facets component

### PR DESCRIPTION
**Pull request by cea2aj**
Create a facets component which renders any facets returned by the search. If any hierarchical facets are returned, they are rendered as standard facets.

J=BACK-2220
TEST=manual, auto

Manually tested this component on the test site and confirmed it rendered all types of facets. Also added a storybook page and automated tests and confirmed they passed.

### CODE REVIEW

File: tests/__utils__/facets.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { DisplayableFacetOption } from '@yext/search-headless-react';
+
+export function getOptionLabelText(option: DisplayableFacetOption) {
```
Comment (EmilyZhang777) [inline tests/__utils__/facets.ts:3]: can you add a an example of what this is trying to match?

File: test-site/src/pages/PeoplePage.tsx [hunk 2]
```diff
@@ -55,11 +55,13 @@ export function PeoplePage() {
             title='Static Employee Department'
             filterOptions={employeeFilterConfigs}
           />
+          <FilterDivider />
           <StaticFilters
             fieldId='c_hierarchicalFacet'
             title='Static Hierarchical Facets'
             filterOptions={hierarchicalFilterConfigs}
           />
+          <FilterDivider />
```
Comment (EmilyZhang777) [inline test-site/src/pages/PeoplePage.tsx:64]: why is this only wrapping StaticFilters and br tag is used on line 73?

**Review by EmilyZhang777 — COMMENTED** [review]
LGTM, just two minor things that mostly are just questions

**Review by benmcginnis — APPROVED** [review]
Lgtm as well, just another minor comment

File: test-site/src/pages/PeoplePage.tsx (hunk 2 above)
Comment (cea2aj) [reply test-site/src/pages/PeoplePage.tsx:64]: These aren't wrapping StaticFilters and are instead standalone filter dividers. The facets dynamically include dividers where needed, but static filters do not

**Pull request merged**